File: BaekJoon/Set/review/Set.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

public class Set {
    public static void main(String args[]) throws Exception{
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));

        int s = 0;
        int m = Integer.parseInt(reader.readLine().trim());

        for (int i = 0; i < m; i++) {
            String cmd = reader.readLine().trim();
            String[] cmdData = cmd.split(" ");
            String op = cmdData[0];

            if (cmdData.length > 1) {
                int num = Integer.parseInt(cmdData[1]);

                switch (op) {
                    case "add":
                        s = s | (1 << num-1);
                        break;
                    case "remove":
                        s = s & ~(1 << num-1);
                        break;
                    case "check":
                        if (((1 << num-1) & s) != 0) {
                            writer.write("1\n");
                        } else {
                            writer.write("0\n");
                        }
                        break;
                    case "toggle":
                        if (((1 << num-1) & s) != 0) {
                            s = s & ~(1 << num-1);
                        } else {
                            s = s | (1 << num-1);
                        }
                        break;
                }
            } else {
                switch (op) {
                    case "all":
                        s = s | (1 << 20) - 1;
                        break;
                    case "empty":
                        s = 0;
                        break;
                }
            }
        }

        writer.flush();
        reader.close();
        writer.close();
    }
}

File: BaekJoon/Set/review/Set.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)

	defer writer.Flush()

	mStr, _ := reader.ReadString('\n')
	mStr = strings.TrimSpace(mStr)
	m, _ := strconv.Atoi(mStr)

	s := make(map[int]struct{})

	for i := 0; i < m; i++ {
		cmd, _ := reader.ReadString('\n')
		cmd = strings.TrimSpace(cmd)
		cmdData := strings.Split(cmd, " ")

		op := cmdData[0]

		if len(cmdData) > 1 {
			num, _ := strconv.Atoi(cmdData[1])

			switch op {
			case "add":
				s[num] = struct{}{}
			case "remove":
				delete(s, num)
			case "check":
				if _, found := s[num]; found {
					fmt.Fprintln(writer, 1)
				} else {
					fmt.Fprintln(writer, 0)
				}
			case "toggle":
				if _, found := s[num]; found {
					delete(s, num)
				} else {
					s[num] = struct{}{}
				}
			}
		} else {
			switch op {
			case "all":
				for j := 0; j < 21; j++ {
					s[j] = struct{}{}
				}
			case "empty":
				s = make(map[int]struct{})
			}
		}
	}
}
